components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2048,47 +2077,77 @@ private int handleStartIndexEqualsNULL(Integer startIndex) {
     }
 
     /**
-     * Get role names according to the given domain. If the domain is not specified, roles of all the user
-     * stores will be returned.
-     *
-     * @param domainName Domain name
-     * @return Roles List
+     * Get role names according to the given domain. If the domain is not specified, roles of all the user stores
+     * will be returned.
+     * @param startIndex
+     * @param count
+     * @param domainName
+     * @return
      * @throws UserStoreException
      * @throws IdentitySCIMException
+     * @throws CharonException
      */
-    private Set<String> getRoleNamesForGroupsEndpoint(String domainName)
-            throws UserStoreException, IdentitySCIMException {
+    private Set<String> getRoleNamesForGroupsEndpoint(int startIndex, Integer count, String domainName)
+            throws UserStoreException, IdentitySCIMException, CharonException {
 
         SCIMGroupHandler groupHandler = new SCIMGroupHandler(carbonUM.getTenantId());
+        Set<String> hybridRoles = new HashSet<>();
+        // If the domain is empty then we are retrieving all the roles and if we have the domain then reteive the
+        // roles from that domain only.
```

Comment:
We should use multiline comments if the comments spans over more than a single line.